File: Unit2-06/Typescript/Jet.ts
/*
 * This program is teh jet class extends from the airplane
 *
 * @author Johanna Liu
 * @version 1.0
 * @since   2024-04-24
 */

import { Airplane } from "./Airplane"

/**
* This is the jet class.
*/
export class Jet extends Airplane {
    /** Jet speed change multiplier. */
    private MULTIPLIER: number = 2

    // Constructor.
    public Jet() {
        // super()
    }

    // Set jet speed.
    public setSpeed(speed: number) {
        super.setSpeed(speed * this.MULTIPLIER)
    }

    // Double speed.
    public accelerate() {
        super.setSpeed(super.getSpeed() * 2)
    }
}

File: Unit2-06/java/Jet.java
/*
* This program is the jet class.
*
* @author  Johanna Liu
* @version 1.0
* @since   2024-03-25
*/

/**
* This is the jet class.
*/
public class Jet extends Airplane {
    /** Jet speed change multiplier. */
    private static final int MULTIPLIER = 2;

    /**
    * Constructor.
    *
    * @param args no arguments
    */
    public Jet(String[] args) {
        // super();
    }

    /**
    * Set jet speed.
    *
    * @param speed set speed
    */
    public void setSpeed(int speed) {
        super.setSpeed(speed * MULTIPLIER);
    }

    /**
    * Double speed.
    *
    * @param args no arguments
    */
    public void accelerate(String[] args) {
        super.setSpeed(super.getSpeed() * 2);
    }
}